### DISCUSSION TITLE
LME correct install folder

### DISCUSSION THREAD

**Discussion by axenstarr** [cisagov/LME#533] [General]
Hello - can I get LME to install in the /opt directory instead of under the root dir? Or is that by design? 

Thanks!

**Comment by aarz-snl** [comment]
Sorry can you expand on what you mean here? install files for lme are copied to /opt/lme during install (the install ansible playbook)

**Reply by axenstarr** [reply]
Sure when I run the command "curl -s https://api.github.com/repos/cisagov/LME/releases/latest | jq -r '.assets[0].browser_download_url' | xargs -I {} sh -c 'curl -L -O {} && unzip -d ~/LME $(basename {})' " It created a LME folder in root not in opt like I thought it would. I'm following the instrucitons from https://github.com/cisagov/LME?tab=readme-ov-file#3-downloading-and-installing-lme

Am I doing something wrong?

**Reply by aarz-snl** [reply]
thats just the install zip. You can install that from anywhere its going to end up at ~/LME of whatever user you're logged into. 

If you dont want it on root logon as a non root user. I.e. "example-user" 

It will add the files to /home/example-user/LME 

During install these files are manipulated to import all configurations, passwords etc and stored in /opt/lme.

The curl command to download the files is the not the install. You're essentially just cloning the files you need to perform the install. Run the curl command and move on to the next steps. By the end of the install you will have all required files in /opt/lme

**Comment by aarz-snl** [comment]
to clarify install should look like:

install all required packages, ansible, etc. 
curl and pull down the zip with install files
run commands to grab hostname ip -- adjust .env file
run the first ansible playbook to install -- wait about 10 minutes for all pods to be up and running.
Look for the post install ansible playbook command -- also included in the readme. Run that command. 

The active install files will exist in /opt/lme at this point

**Reply by axenstarr** [reply]
Awesome thank you!

**Reply by axenstarr** [reply]
I do have one more question. I am not getting this directory (lme-2-arch) after I run the first command. So I dont have a lme-2-arch to goto before I run the install_lme command. I ran the first command (sudo apt update && sudo apt install -y ansible) so not sure if that lme-2-arch dir is suppoed to be created or not.

You can run this Ansible installer for a fully automated install.
sudo apt update && sudo apt install -y ansible
#cd ~/LME/lme-2-arch # Or path to your clone of this repo
ansible-playbook ./ansible/install_lme_local.yml

**Reply by aarz-snl** [reply]
You dont have to be in lme-2-arch -- I'm not really sure what that is -- may be old directions. 

Earlier in the directions you go to: 

cd ~/LME 

Just run all commands from this LME path -- follow the 2nd part of the comment where it says "Or path to your clone of this repo" which should be ~/LME

**Reply by axenstarr** [reply]
Thanks so now I am at the posts-install and I am not able to connect to kibana https://ipofserver:8080 and I also dont have a post_install_local.yml that I can find in the lme folder or config (ansible-playbook ./ansible/post_install_local.yml)

**Reply by axenstarr** [reply]
Ok I see its going back to ~/LME  and that seemed to work when I ran the (ansible-playbook ./ansible/post_install_local.yml command. I'll try the kibana url again after its done.

**Comment by axenstarr** [comment]
Any advice on fixing why dasboards did not finish? I also have bunch of errors during the post install for the dashboards.

root@canvlog01:/opt/lme# sudo -i ls -al /opt/lme/dashboards/elastic/INSTALLED
ls: cannot access '/opt/lme/dashboards/elastic/INSTALLED': No such file or directory
root@canvlog01:/opt/lme# sudo -i ls -al /opt/lme/dashboards/wazuh/INSTALLED
ls: cannot access '/opt/lme/dashboards/wazuh/INSTALLED': No such file or directory


Dashboard errors:
failed: [localhost] (item=/opt/lme/dashboards/elastic/computer_software_overview_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/computer_software_overview_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/security_dashboard_security_log_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/security_dashboard_security_log_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/privileged_activity_log_dashboards_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/privileged_activity_log_dashboards_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/policy_changes_and_system_activity_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/policy_changes_and_system_activity_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/user_security_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/user_security_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/credential_access_logs_dashboard_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/credential_access_logs_dashboard_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/alerting_dashboard_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/alerting_dashboard_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/healthcheck_dashboard_overview_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/healthcheck_dashboard_overview_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/process_explorer_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/process_explorer_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/sysmon_summary_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/sysmon_summary_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/user_hr_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/user_hr_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}
failed: [localhost] (item=/opt/lme/dashboards/elastic/identity_access_management_2_0.ndjson) => {"ansible_loop_var": "item", "changed": false, "item": "/opt/lme/dashboards/elastic/identity_access_management_2_0.ndjson", "msg": "Unsupported parameters for (ansible.legacy.command) module: warn. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends."}

**Reply by axenstarr** [reply]
Looking at this it looks like Kibana is unreachable and I am not sure how to fix that. if I just goto http://myipaddress I get the elastic webpage.

**Comment by axenstarr** [comment]
I was wondering if I could get some help with the dashboard issues? I also cannot get to http://ipofserver:8080 for Kibana so not sure what is up with that. I do notice that 8080 is not listening but I also see the firewall inactive.  

If I goto just the IP address for my LME server I get the elastic page but no password works so not sure how to login there.

**Comment by aarz-snl** [comment]
i'd recommending doing an uninstall and start over with a clean install.  Follow instructions closely -- you shouldn't need to go to port 8080 -- kibana is 443 -- so its going to be https://ipofserver -- there will be no http (non secure) apps you will connect to with this install.

Make sure ansible playbooks run all the way through and every file is created in /opt/lme

If you have issues where its saying something doesn't exist but it does exist then you have permissions issues. You may need to manually set them and then run playbooks again. The playbooks set permissions I'm only saying that a troubleshooting step

passwords can be grabbed by running ./scripts/extract_secrets.sh -p from ~/LME -- try not to deviate from the instructions as all steps are from certain folders its expecting to exist.

Take your time on each step and wait 10 minutes in between running first playbook and post install playbook. Make sure every container has been up and running for a couple minutes before running post install. Make sure you read the entire README.md closely

its hard to help on my end because it can be any number of issues.

**Reply by axenstarr** [reply]
I completly understand. I'll run the uninstall and go over it again and thanks for replying.

**Comment by axenstarr** [comment]
Hello - So I am re-installing and like last time I am getting this for the services. Are the generated ones supposed to be like that?

![image](https://github.com/user-attachments/assets/67ce6dc1-18c6-4e76-a91a-c29c616d24a8)

The services look ok so the above screen shot has me confused.

![image](https://github.com/user-attachments/assets/b9bb072c-d192-4613-aa07-6d2452889f6f)

**Comment by axenstarr** [comment]
So I am back to where ai was before but now atleast I was able to see the passwords. I missed the source passwords part before. Where I am still stuck at is I cannot get to my Kibana at all. I tried ssh into it but I cannot connect via password. I tried the https site and still no site available. I know I followed everything up to this point properly. I can goto my IP address and get to the elastic page and that password works so I have no reason to think the other passwords are bad.

#connect via ssh if you need to 
ssh -L 8080:localhost:5601 x.x.x.x (password does not seem to work)
#go to browser:
#https://localhost:8080

**Comment by axenstarr** [comment]
Ok I finally got it working with some help. The error we found when I had those dashboard errors were here:

That error says it won’t run the ansible command because there’s deprecated language in it. Probably should report an issue for it on GitHub.
 
You could edit post_install_local.yml and remove the args: and warn: lines from the commands to upload the dashboards. 
 
The line numbers from the post_install_local.yml that were causing it to blow are in the screen shot. If I need to post a bug let me know.
 
![image](https://github.com/user-attachments/assets/73237e61-7b7a-442b-a177-33d49f8cae66)
